### PR TITLE
http refactor

### LINKED ISSUE #44: Handle ClientConnectorError in connector
Exception text: `ClientConnectorError: Cannot connect to host dynamodb.<snip>.amazonaws.com:443 ssl:default [None]`

Exception type: `aiohttp.client_exceptions.ClientConnectorError`

Stack:
```py
aiohttp/connector.py in _wrap_create_connection at line 943
aiohttp/connector.py in _create_direct_connection at line 980
aiohttp/connector.py in _create_direct_connection at line 1004
aiohttp/connector.py in _create_connection at line 858
aiohttp/connector.py in connect at line 523
aiohttp/client.py in _request at line 480
aiohttp/client.py in __aenter__ at line 1012
aiodynamo/http/aiohttp.py in post at line 31
aiodynamo/client.py in send_request at line 651
aiodynamo/client.py in get_item at line 454
aiodynamo/client.py in get_item at line 124
```

### LINKED ISSUE #45: question: connector get/post parity
How come `get` has a `timeout=` mandatory keyword argument, but `post` does not have a `timeout=` argument?

### LINKED ISSUE #102: Auto paginating endpoints get throttled if passed invalid value
For example if pass limit `"foobar"` to `limit` in count it gets throttled. At least in tests against DynamoDb Local

```
____________________________________________________________________________________________________________ test_count_with_limit[httpx-True] _____________________________________________________________________________________________________________

client = Client(http=HTTPX(client=<httpx.AsyncClient object at 0x105f3f4f0>), credentials=<aiodynamo.credentials.ChainCredentia...class 'float'>, throttle_config=ExponentialBackoffThrottling(time_limit_secs=60, base_delay_secs=2, max_delay_secs=20))
prefilled_table = '429ba920-a8ec-44b1-bd2b-44c17b4ee4b4', consistent_read = True

    @pytest.mark.parametrize("consistent_read", [None, True, False])
    async def test_count_with_limit(
        client: Client, prefilled_table: TableName, consistent_read: Optional[bool]
    ) -> None:
>       assert (
            await client.count(
                prefilled_table,
                HashKey("h", "h"),
                limit="foobar", #90,
                consistent_read=consistent_read,
            )
            == 90
        )

tests/integration/test_client.py:150: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
src/aiodynamo/client.py:824: in count
    async for result in self._depaginate("Query", payload, limit):
src/aiodynamo/client.py:999: in _depaginate
    result = await task
src/aiodynamo/client.py:945: in send_request
    async for _ in self.throttle_config.attempts():
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = ExponentialBackoffThrottling(time_limit_secs=60, base_delay_secs=2, max_delay_secs=20)

    async def attempts(self) -> AsyncIterable[None]:
        deadline = time.monotonic() + self.time_limit_secs
        for delay in self.delays():
            yield
            if time.monotonic() > deadline:
>               raise Throttled()
E               aiodynamo.errors.Throttled

src/aiodynamo/models.py:238: Throttled
```

### LINKED ISSUE #109: Http wrapping issues
When I was adding credentials handler to [assume role with web identity](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithWebIdentity.html) which we need for our EKS setup I came across few issues why I am for now implementing it as custom code in our own project instead of providing it for the library.

Http wrappers assume response is json, https://github.com/HENNGE/aiodynamo/blob/master/src/aiodynamo/http/base.py but as endpoint response is xml (and at least I can't find any mention of possibility to request it as json, but you know AWS docs 💩 ). This is blocker and thought that perhaps there needs to be some refactoring around http handling which takes too long for us.

Some additional things I noticed
1) in `MetadataCredentials` there is [fetch_with_retry](https://github.com/HENNGE/aiodynamo/blob/master/src/aiodynamo/credentials.py#L236) which couldn't use because it assumes `GET` but endpoint for assume role is `POST`. Was thinking should/could this retry option be in base http classes?

2) Missing timeout in `POST`, would prefer to have timeout also for this credentials call.  As mentioned in https://github.com/HENNGE/aiodynamo/issues/45

3) http.post(..) requires body as bytes even when it can in aiohttp and httpx also be None. https://github.com/HENNGE/aiodynamo/blob/master/src/aiodynamo/http/base.py#L28-L30

### PR DESCRIPTION
**Pull request by ojii**
Fixes #45
Fixes #102
Fixes #44
Fixes #109

### CODE REVIEW

**Comment by dimaqq** [comment]
re tests: probably an event loop fixture is needed 🤔 
or maybe it's newer `pytest-asyncio`?

**Review by dimaqq — APPROVED** [review]
Yay, 2022, a chance to make "breaking" changes :)
Consider adding a mapping old/new names to the changelog.
I think it's a most visible in the tests right in this PR.

**Comment by ojii** [comment]
> re tests: probably an event loop fixture is needed 🤔 or maybe it's newer `pytest-asyncio`?

it was the newer pytest-asyncio which meant the previous way I ran session-scoped async fixtures didn't work anymore, switch it to a better one.

File: src/aiodynamo/http/httpx.py [hunk 1]
```diff
@@ -1,51 +1,24 @@
...
-                str(url), content=body, headers=headers or {}
+class HTTPX:
+    client: httpx.AsyncClient
+
+    async def __call__(self, request: Request) -> Response:
+        try:
+            response = await self.client.request(
+                method=request.method,
+                url=request.url,
+                # httpx is coded with no_implicit_optionals=false, we use strict=true
```
Comment (dimaqq) [inline src/aiodynamo/http/httpx.py:18]:
```suggestion
                # httpx is coded with no_implicit_optional=False, we use strict=True
```

I think that's the `mypy` spelling ☝️

File: docs/concepts.rst [hunk 2]
```diff
@@ -85,3 +85,9 @@ Compare the following two calls::
             write=2
         )
     )
+
+HTTP Client Library independence
```
Comment (dimaqq) [inline docs/concepts.rst:89]:
```suggestion
HTTP Client Library Independence
```
for consistency maybe?

Comment (dimaqq) [reply docs/concepts.rst:89]: But then again title case is all over the place in `advanced.rst` so 🤷🏿‍♂️

**Review by dimaqq — APPROVED** [review]

**Pull request merged**